docs/tutorials/services/constrain-motion.md

```diff
@@ -0,0 +1,432 @@
+{
+  "components": [
+    {
+      "name": "myArm",
+      "type": "arm",
+      "model": "xArm6",
+      "attributes": {
+        "host": "<ip-address>",
+        "port": 0,
+        "speed_degs_per_sec": 15,
+        "acceleration_degs_per_sec_per_sec": 0
+      },
+      "depends_on": [],
+      "frame": {
+        "parent": "world",
+        "translation": {
+          "x": 0,
+          "y": 0,
+          "z": 0
+        },
+        "orientation": {
+          "type": "ov_degrees",
+          "value": {
+            "x": 0,
+            "y": 0,
+            "z": 1,
+            "th": 0
+          }
+        }
+      }
+    },
+    {
+      "name": "myGripper",
+      "type": "gripper",
+      "model": "fake",
+      "attributes": {},
+      "depends_on": [
+        "myArm"
+      ],
+      "frame": {
+        "parent": "myArm",
+        "translation": {
+          "x": 0,
+          "y": 0,
+          "z": 0
+        },
+        "orientation": {
+          "type": "ov_degrees",
+          "value": {
+            "x": 0,
+            "y": 0,
+            "z": 1,
+            "th": 0
+          }
+        },
+        "geometry": {
+          "x": 73,
+          "y": 73,
+          "z": 90,
+          "translation": {
+            "x": 0,
+            "y": 0,
+            "z": 0
+          }
+        }
+      }
+    }
+  ]
+}
+```
+
+{{% /expand %}}
+
+<!--
+## Visualize your table
+
+In the previous tutorial you [created a representation of a table](../plan-motion-with-arm-gripper/#describe-the-robots-working-environment) in your client code.
+You will use this same code later in this tutorial.
+
+Since this tutorial gets a bit more complicated than the last, let's configure a representation of the table so you can see it in the Frame System visualizer.
+This configured table won't be taken into account by the Motion Service, but it's useful to be able to see it.
+
+On your robot's **Config** tab, create a new component called `table` with **Type** `generic` and **Model** `fake`.
+Click **Create component**, then click **Add frame**.
+
+Go to the **Frame System** subtab.
+Click **table** on the left menu.
+For **Geometry** click **box**.
+
+Set the dimensions (**Size**) to `1000` x `1000` x `20`.
+Notice that the top of the table is now 10mm above the origin, because the 20mm thickness is centered on the centerpoint.
+Account for this by giving the table a **Translation** of `-10` in the Z direction, moving the top of the table to 0.
+
+![The Frame System subtab of the Components tab with a 1000 wide, 1000 deep, 200 millimeter thick box representing the top of the table.](../../img/constrain-motion/frame-table.jpg)
+
+### Visualize obstacles on the table
+-->
+
+## Modify your robot's working environment
+
+In the last tutorial, you [defined a table obstacle](../plan-motion-with-arm-gripper/#describe-the-robots-working-environment) to prevent the arm and gripper from hitting the table upon which the arm is mounted.
```

Comment:
```suggestion
In the previous tutorial, you [defined a table obstacle](../plan-motion-with-arm-gripper/#describe-the-robots-working-environment) to prevent the arm and gripper from hitting the table upon which the arm is mounted.
```